src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1136,87 +1224,79 @@ void set_signal_handler(int sig, bool set_installed) {
   assert(oldhand2 == oldhand, "no concurrent signal handler installation");
 }
 
-// install signal handlers for signals that HotSpot needs to
-// handle in order to support Java-level exception handling.
-
-bool PosixSignals::are_signal_handlers_installed() {
-  return signal_handlers_are_installed;
-}
-
 // install signal handlers for signals that HotSpot needs to
 // handle in order to support Java-level exception handling.
 void PosixSignals::install_signal_handlers() {
-  if (!signal_handlers_are_installed) {
-    signal_handlers_are_installed = true;
-
-    // signal-chaining
-    typedef void (*signal_setting_t)();
-    signal_setting_t begin_signal_setting = NULL;
-    signal_setting_t end_signal_setting = NULL;
-    begin_signal_setting = CAST_TO_FN_PTR(signal_setting_t,
-                                          dlsym(RTLD_DEFAULT, "JVM_begin_signal_setting"));
-    if (begin_signal_setting != NULL) {
-      end_signal_setting = CAST_TO_FN_PTR(signal_setting_t,
-                                          dlsym(RTLD_DEFAULT, "JVM_end_signal_setting"));
-      get_signal_action = CAST_TO_FN_PTR(get_signal_t,
-                                         dlsym(RTLD_DEFAULT, "JVM_get_signal_action"));
-      libjsig_is_loaded = true;
-      assert(UseSignalChaining, "should enable signal-chaining");
-    }
-    if (libjsig_is_loaded) {
-      // Tell libjsig jvm is setting signal handlers
-      (*begin_signal_setting)();
-    }
+  static bool done = false;
+  assert(!done, "Only call once");
```

Comment:
Not sure this kind of init guard is really needed, but if it is then the flag should DEBUG_ONLY. Also I don't see you setting this to true anywhere.

Reply:
Ouch. Stupid rebasing error.